lib/plugins.js

```diff
@@ -116,9 +116,11 @@ module.exports = class Plugins {
     }
 
     // Register, log, return
-    this.registry.push(plugin);
-    this.log.debug('plugin %s loaded from %s', plugin.name, file);
-    this.log.silly('plugin %s has', plugin.name, plugin.data);
+    if (!this.registry.find(p => p.name === plugin.name)) {
```

Comment:
this can be further optimized so that we don't do this O(n) find but I guess it's good enough :)